org.eclipse.gef/src/org/eclipse/gef/internal/ui/rulers/DragGuidePolicy.java

```diff
@@ -35,11 +34,12 @@
  */
 public class DragGuidePolicy extends GraphicalEditPolicy {
 
-	private List attachedEditParts = null;
-	private IFigure dummyGuideFigure, dummyLineFigure;
+	private List<? extends EditPart> attachedEditParts = null;
+	private IFigure dummyGuideFigure;
+	private IFigure dummyLineFigure;
 	private boolean dragInProgress = false;
 
-	protected IFigure createDummyLineFigure() {
```

Comment:
I assume this method used to be marked as protected, to be overwritten by subclasses? As far as I can tell, this method is only referenced by `getDummyLineFigure()`. Wouldn't it make more sense to simply inline this method? Perhaps it would even make sense to do the same for `createDummyGuideFigure()`...

You already did the same in the RulerEditPartFactory, might as well apply it here, too.

Reply:
I guess this came from the general design pattern in GEF Classic that subclasses can with such methods easily provide their own figures. However as the class is internal and I don't see different policies in the near future soon I think it is better to inline them as you suggested.